src/CommandLineUtils/Utilities/Prompt.OptionsManager.cs

```diff
@@ -0,0 +1,378 @@
+                    {
+                        Boxes.Add(new OptionsOption(property.Name));
+                    }
+                }
+                else
+                {
+                    foreach (var property in type.GetTypeInfo()
+                        .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                        .Where(i => i.PropertyType == typeof(bool) &&
+                                    i.GetCustomAttribute<UseOnOptionsAttribute>() != null))
+                    {
+                        Boxes.Add(new OptionsOption(property.Name));
+                    }
+                }
+
+                Redraw();
+            }
+
+            private void BoxesOnCollectionChanged(object sender,
+                NotifyCollectionChangedEventArgs notifyCollectionChangedEventArgs)
+            {
+                foreach (var item in notifyCollectionChangedEventArgs.NewItems)
+                {
+                    if (item is OptionsOption checkboxSelection)
+                    {
+                        checkboxSelection.PropertyChanged += (o, args) =>
+                        {
+                            if (args.PropertyName != nameof(OptionsOption.IsSelected))
+                            {
+                                return;
+                            }
+
+                            SetPropertyValue(checkboxSelection.Title, checkboxSelection.IsSelected);
+                            Redraw();
+                        };
+                    }
+                }
+            }
+
+            private void SetPropertyValue(string name, object value)
+            {
+                Model?.GetType().GetTypeInfo().GetProperty(name).SetValue(Model, value);
+            }
+
+            private void GetPropertyValue(string name, object value)
+            {
+                Model?.GetType().GetTypeInfo().GetProperty(name).GetValue(value);
+            }
+
+            private static void ClearConsoleRange(int startY, int endY, int x = -1)
+            {
+                if (x < 0)
+                {
+                    x = Console.WindowWidth;
+                }
+
+                for (var i = startY; i < endY; i++)
+                {
+                    ClearConsoleLine(i, x);
+                }
+            }
+
+            private static void ClearConsoleLine(int line, int length)
+            {
+                WriteTemporarly(() =>
+                {
+                    Console.SetCursorPosition(0, Console.CursorTop);
+                    Console.Write(new string(' ', length));
+                }, 0, line);
+            }
+
+            /// <summary>
+            /// Shows this instance.
+            /// </summary>
+            public void Show()
+            {
+                Console.CursorVisible = false;
+
+                while (true)
+                {
+                    var key = Console.ReadKey();
+                    if (Options.Keys.Finalize.Contains(key.Key))
```

Comment:
I don't think we need this right now, but it might be nice to support using ctrl or alt to jump to the bottom or top of a list. For now, ignoring input when modifiers are used seems simpler.